src/org/pentaho/metadata/util/Util.java

```diff
@@ -99,14 +86,25 @@ public static boolean validateId( CharSequence id ) {
 
     for ( int i = 0, len = id.length(); i < len; i++ ) {
       char ch = id.charAt( i );
-      if ( !isLatinLetter( ch ) && !isAsciiDigit( ch ) && !isAcceptableNonLetter( ch ) ) {
+      if ( isUnacceptableCharacter( ch ) ) {
         return false;
       }
     }
 
     return true;
   }
 
+  /**
+   * Check if character is unacceptable for MQL and need to be converted.
+   * List of unacceptable characters should corresponds to the regexp's inside the Util#toId method.
+   *
+   * @param ch character to check
+   * @return true if character is unacceptable for MQL, false otherwise
+   */
+  private static boolean isUnacceptableCharacter( char ch ) {
+    return " .,:(){}[]\"`'*/+-".indexOf( ch ) != -1;
```

Comment:
Let's raise this up to a public constant
public static final MQL_RESERVED_CHARS = " .,:(){}[]\"`'*/+-";